cmd/copy.go

```diff
@@ -948,6 +1025,136 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	return nil
 }
 
+// TODO: include this within initResourceTraverser?
+func initPipeline(ctx context.Context, location common.Location, credential common.CredentialInfo) (p pipeline.Pipeline, err error) {
+	switch location {
+	case common.ELocation.Local():
+		return nil, nil
+	case common.ELocation.Blob():
+		p, err = createBlobPipeline(ctx, credential)
+	case common.ELocation.File():
+		p, err = createFilePipeline(ctx, credential)
+	case common.ELocation.BlobFS():
+		p, err = createBlobFSPipeline(ctx, credential)
+	default:
+		err = fmt.Errorf("can't produce new pipeline for location %s", location)
+	}
+
+	return
+}
+
+func trimWildcards(path string) string {
```

Comment:
For remote sources, yeah, it will chop off the trailing /*. For local sources, we support wildcards in the path, so it will chop off ALL wildcards. It should *probably* be changed to getPathBeforeFirstWildcard.